pkg/controller/backupstoragelocation_controller.go

```diff
@@ -0,0 +1,147 @@
+/*
```

Comment:
I was thinking the controllers could go into an `internal/controller` package. Is that breaking some sort of controller-runtime pattern? My reasoning is that they're internal to Velero's operation and would never be used by Go API clients, only ever being used by submitting CRDs to the Kube API server.

If this is incompatible with controller-runtime practices though, feel free to ignore.

Reply:
This is very subjective and depending, I think, on what we want to accomplish. If we want to accomplish consistency with other kubebuilder generated projects, then I'd suggest leaving it here. In fact, kubebuilder generates all controllers under a directory `controllers` (plural), and at the root, so once we convert all controllers we should move them to that structure.

Reply:
But other than consistency there's no requirement to have the controllers be public. I'm not sure how we should optimize this, I'm tending to go with consistency (with other kubebuilder projs).

Reply:
Ok, that makes sense.

I was envisioning that eventually we moved all the controllers into `internal/controller(s?)`. However, that's not a hard requirement. If it's convention to keep them public, then I'm not going to push the issue. They're not generally useful to import, in any case.